### PR TITLE
adding flatMap support

### PR DESCRIPTION
**Pull request by kfiron**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,12 @@ provide a number of functions that help with manipulating optional values.
 * `some(value).map(func)` returns `some(func(value))`
 * `none.map(func)` returns `none`
 
+### flatMap(*func*)
+
+* `some(value).map(func)` returns `func(value)`
+* `none.map(func)` returns `none`
```
Comment (mwilliamson) [inline README.md]: I think these should be calls to `flatMap`.

File: test/options.test.js [hunk 2]
```diff
@@ -67,6 +67,33 @@ exports.mappingOverSomeAppliesFunctionToValue = function(test) {
     test.done();
 };
 
+
+exports.flatMapOnNoneToNone = function(test) {
```
Comment (mwilliamson) [inline test/options.test.js:71]: I wonder if just putting some pseudo-JavaScript would make the test clearer e.g.

``` javascript
exports["none.flatMap(func) is none"] = ...
```

**Comment by mwilliamson** [comment]
Thanks for the pull request, I left a couple of thoughts.

File: test/options.test.js (hunk 2 above)
Comment (kfiron) [inline test/options.test.js:71]: I am not sure i understand what you mean, can you elaborate?

**Comment by kfiron** [comment]
I have fixed the README.md, i don't have clear idea about how to make this test clearer, i didn't understand your advice, can you help?

File: test/options.test.js (hunk 2 above)
Comment (mwilliamson) [inline test/options.test.js:71]: I'm suggesting that the existing test names aren't especially clear, and that writing in the style of the example I wrote above might be clearer.

Comment (kfiron) [inline test/options.test.js:71]: Ok, I will change later, thanks :)

On Tuesday, 5 January 2016, Michael Williamson notifications@github.com
wrote:

> In test/options.test.js
> https://github.com/mwilliamson/node-options/pull/8#discussion_r48879985:
> 
> > @@ -67,6 +67,33 @@ exports.mappingOverSomeAppliesFunctionToValue = function(test) {
> >      test.done();
> >  };
> > 
> > +
> > +exports.flatMapOnNoneToNone = function(test) {
> 
> I'm suggesting that the existing test names aren't especially clear, and
> that writing in the style of the example I wrote above might be clearer.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mwilliamson/node-options/pull/8/files#r48879985.

## 

_Kfir Bloch_

Cell: 054-568-5848

40 Hanamal street, Tel Aviv, Israel

_"At some point you are going to lose consistency"_

**Comment by mwilliamson** [comment]
The commits have been squashed and merged in, and the changes released as 0.2.3. Thanks again for the pull request.

**Pull request closed without merging**